build/dangerjs/detectNonApprovedDependencies/utils/checkPackageVersionConstraints.ts

```diff
@@ -0,0 +1,62 @@
+import * as semver from 'semver'
+
+import approvedPackages, { isIgnored } from '../approvedPackages'
+import { getPackageName, getPackageVersion } from './packageNameUtils'
+
+export type FailedConstraintsExplanation = {
+  approvedPackages: string[]
+  failedConstraints: string[]
+}
+
+export type ApprovalCheckResult = {
+  isApproved: boolean
+  explain?: FailedConstraintsExplanation
+}
+
+const checkPackageVersionConstraints = (
+  packageId: string,
+  versionConstraints: string[],
+): ApprovalCheckResult => {
+  const packageName = getPackageName(packageId)
+
+  if (isIgnored(packageId)) {
+    return { isApproved: true }
+  }
+
+  const versionConstraintsCheckResult = versionConstraints.reduce(
+    (acc, versionConstraint) => ({ ...acc, [versionConstraint]: false }),
+    {},
+  )
+
+  const approvedPackageIds = approvedPackages.filter(item => item.startsWith(`${packageName}@`))
+
+  approvedPackageIds.map(getPackageVersion).forEach(approvedPackageVersion => {
+    Object.keys(versionConstraintsCheckResult).forEach(versionConstraint => {
+      if (semver.satisfies(approvedPackageVersion, versionConstraint)) {
+        versionConstraintsCheckResult[versionConstraint] = true
+      }
+    })
+  })
```

Comment:
It seems that there is a bug with `webpack` at least:

```
versionConstraintsCheckResult webpack {}
approvedPackageIds [ 'webpack@4.25.1' ]
approvedPackageIds.map(getPackageVersion) [ '4.25.1' ]
versionConstraintsCheckResult webpack {}
```

It's safe to ignore for Webpack, but can you please check other packages?

Reply:
discussed offline, agreed that logic of the approvals check handles the case correctly